addons/stock/report/report_stock_reception.py

```diff
@@ -0,0 +1,285 @@
+        for source, dummy in sources_to_lines.items():
+            sources_to_formatted_scheduled_date[source] = self._get_formatted_scheduled_date(source)
+
+        return {
+            'data': data,
+            'doc_ids': docids,
+            'doc_model': 'stock.picking',
+            'sources_to_lines': sources_to_lines,
+            'precision': self.env['decimal.precision'].precision_get('Product Unit of Measure'),
+            'pickings': pickings,
+            'sources_to_formatted_scheduled_date': sources_to_formatted_scheduled_date,
+        }
+
+    def _prepare_report_line(self, quantity, product, move_out, source=False, is_assigned=False, is_qty_assignable=True, move_ins=False):
+        return {
+            'source': source,
+            'product': {
+                'id': product.id,
+                'display_name': product.display_name
+            },
+            'uom': product.uom_id.display_name,
+            'quantity': quantity,
+            'is_qty_assignable': is_qty_assignable,
+            'move_out': move_out,
+            'is_assigned': is_assigned,
+            'move_ins': move_ins and move_ins.ids or False,
+        }
+
+    def _get_formatted_scheduled_date(self, source):
+        """ Unfortunately different source record types have different field names for their "Scheduled Date"
+        Therefore an extendable method is needed.
+        """
+        if source._name == 'stock.picking':
+            return format_date(self.env, source.scheduled_date)
+        return False
+
+    def action_assign(self, move_ids, qtys, in_ids):
+        """ Links picking move(s) to the next step (move_ids in another picking)
+        """
+
+        def _link_move(out, qty_to_link, ins):
+            out = self.env['stock.move'].browse(out)
+            potential_ins = self.env['stock.move'].browse(ins)
+
+            if float_compare(out.product_qty, qty_to_link, precision_rounding=out.product_id.uom_id.rounding) == 1:
+                # we are only assigning part of its demand => let's split it now to prevent reservation problems later
+                new_move_vals = out._split(out.product_qty - qty_to_link)
+                new_out = self.env['stock.move'].create(new_move_vals)
+                new_out._action_confirm(merge=False)
+                if potential_ins[0].state != 'done' and out.reserved_availability:
+                    out.move_line_ids.move_id = new_out
+                elif potential_ins[0].state == 'done' and out.reserved_availability > qty_to_link:
+                    # we need to make sure we don't accidentally mess up the reservable amount if out is unreserved + re-reserved at some point
+                    out.move_line_ids.move_id = new_out
+                    assigned_amount = 0
+                    for move_line_id in new_out.move_line_ids:
+                        if assigned_amount + move_line_id.product_qty > qty_to_link:
+                            new_move_line = move_line_id.copy()
```

Comment:
never copy a stock.move.line atm. product_qty is copy but the quant is not update. Gonna push a fix for stable but use product_qty, product_uom_qty to zero in copy vals

Reply:
Ah, got it. Set the default copy values for `product_qty` = `product_uom_qty` = `qty_done`. I also realized I forgot the `qty_done` != 0 case :sob: hopefully it's ok to leave all qty done on the original move line for now?